### ISSUE TITLE
[sequoia beta] ui issue: all keys are marked as "locked"

### ISSUE THREAD

**Issue by martinpaljak** [maxgoedjen/secretive#560]
All keys in the key list have a padlock, even the ones that are functioning as "notify only"

**Comment by maxgoedjen** [comment]
Yeah, I can also reproduce this. Not clear to me yet if it's a Sequoia bug or just a change in some underlying behavior I'll have to update for.

**Comment by maxgoedjen** [comment]
https://github.com/maxgoedjen/secretive/blob/main/Sources/Packages/Sources/SecureEnclaveSecretKit/SecureEnclaveStore.swift#L272

This is where it's happening. Basically there's not a _proper_ way to inspect for the auth-required attribute, so I have been comparing them with a reference. Something in the construction of this object must've changed between versions though, giving us these false positives. If I can figure out a way to fix this soonish, I"ll do that, but I may just have to remove this feature :/

**Comment by maxgoedjen** [comment]
Good news/bad news on this – I don't think there's going to be a great way to fix this for older secrets. Good news is the new improvements in the store coming in 3.0 track this explicitly so they should not be affected.

**Comment by martinpaljak** [comment]
Well, it is a good reason to actually do regular secrets rotation ;)

**Comment by maxgoedjen** [comment]
The migrator in 3.0 makes a best guess at this – but unfortunately at the moment I don't have any keys that are in that state currently.

https://github.com/maxgoedjen/secretive/blob/main/Sources/Packages/Sources/SecureEnclaveSecretKit/CryptoKitMigrator.swift#L45-L47
